tutorial/practicals/LFRic/building_code/4_psydata/README.md

```diff
@@ -0,0 +1,227 @@
+way you like. The options are provided as an additional dictionary argument to the ``apply``
+function of a transformation, with "region_name" as key, and then a tuple containing first a module name, then a
+local name. You could use:
+
+    {"region_name": ("time_evolution", "propagate")}
+
+Add this dictionary to your transformation script, and rebuild your application. You
+have to do a
+
+    make -f Makefile.extract clean
+
+since the Makefile does not have a dependency on your script (though it would be
+recommended to add this to the Makefile so that a change in your script automatically
+triggers running PSyclone again).
+
+You should now see that the psy-layer contains:
+
+```fortran
+    CALL extract_psy_data%PreStart("time_evolution", "propagate", 7, 2)
+```
+
+I.e. it uses the names you have provided, and running the binary will now create
+a NetCDF file called ``time_evolution-propagate.nc``.
+
+## Step 6: Examining the output
+
+The NetCDF command ``ncdump`` can be used to check the content of the created file:
+
+```bash
+    ncdump time_evolution-propagate.nc | less
+```
+
+You will see that the dimensions are defined for the fields used. Some variables
+are listed with the postfix ``post``, e.g. ``perturbation_post``. These are the
+values of the variables *after* the kernel call, so it allows the NetCDF file
+to store perturbation as input parameter (just ``perturbation``), and its
+values after the kernel (``perturbation_post``).
+
+At this stage no driver is created that can read in the file.
+
+
+## Step 7: Instrument more than one invoke
+
+The transformation script can be slightly changed to instrument all invokes
+in a file. While this might not be useful for kernel extraction, it is
+essential for e.g. parameter verification, where you typically want to
+check all invokes contained in a program. Use the template ``transform_all.py``
+to apply the extraction transformation to all invokes. This script actually
+requires less changes than the ``transform_one.py`` template (since it
+works on all )
+Following the same process, just using this more general transformation script,
+will result in a binary that creates two NetCDF files at run time.
+
+
+## Step 8: Try other PSyData libraries
+The following set of PSyData libraries is available and can be tested.
+Corresponding Makefiles are provided.
+
+### Readonly Verification
+This library makes sure that kernel arguments that are declared as read-only (in metadata) are not modified. The compiler
+should make sure that a variable declared with ``intent(in)`` is not explicitly
+overwritten, but a memory overwrite because of an out-of-bound access to a
+different array is always possible. To import the transformation use:
+
+```python
+    from psyclone.psyir.transformations import ReadOnlyVerifyTrans
+```
+
+Then create a transformation script based on ``transform_all.py`` script to
+instrument your application. If you run this script, nothing seems to happen.
```

Comment:
Point out that they need to use `make` and a different makefile. Also, can this transformation be applied to a whole schedule?

Reply:
I have renamed all Makefiles and transformations to make them all more consistent. For each ``XXX_transform.py`` is now a corresponding ``Makefile.XXX``. While this might be a bit of an overkill (de-facto all Makefile.XXX are identical except for the paths, same for the transforms), but it should minimise confusion when running the tutorial. And of course same files in the ``solutionds`` directory.

I've also updated the README.me to reference the right transform or Makefile everywhere now.